cmd/kube-controller-manager/app/plugins_providers.go

```diff
@@ -28,11 +32,24 @@ import (
 	"k8s.io/kubernetes/pkg/volume/vsphere_volume"
 )
 
+type probeFn func() []volume.VolumePlugin
+
+func appendPlugin(plugins []volume.VolumePlugin, key featuregate.Feature, fn probeFn) []volume.VolumePlugin {
+	// Skip appending the in-tree plugin to the list of plugins to be probed/initialized
+	// if the CSIMigrationInTreeOff feature flag and plugin specific CSI migration
+	// feature flag is enabled
+	if feature.DefaultFeatureGate.Enabled(features.CSIMigrationInTreeOff) && feature.DefaultFeatureGate.Enabled(key) {
+		return plugins
```

Comment:
Would it be useful to klog a message saying we are "skipping in-tree plugin init, because..."?

Reply:
Ditto for kubelet and the other copies of this.

Reply:
Done